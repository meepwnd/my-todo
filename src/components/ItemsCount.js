import React from "react";
import { connect } from "react-redux";

const ItemsCount = props => {
  const { completedTodosCount, todosCount } = props;
  const activeCount = todosCount - completedTodosCount;
  const itemWord = activeCount === 1 ? "item" : "items";

  return (
    <p className="item-count">
      {activeCount} {itemWord} left
    </p>
  );
};

const getCompletedTodosCount = todos =>
  todos.reduce((count, todo) => (todo.completed ? count + 1 : count), 0);

const mapStateToProps = state => ({
  completedTodosCount: getCompletedTodosCount(state.options),
  todosCount: state.options.length
});

export default connect(mapStateToProps)(ItemsCount);
