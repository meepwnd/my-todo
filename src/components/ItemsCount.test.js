import React from 'react';
import {shallow} from 'enzyme'
import {ItemsCount} from './ItemsCount'

describe('itemscout component with no items', () => {
  it ('should render correctly', () => {
    const wrapper = shallow(<ItemsCount />)
    expect(wrapper).toMatchSnapshot()
  })

  it ('should render correctly when 1 item left', () => {
    const completedTodosCount = 2;
    const todosCount = 3;
    const wrapper = shallow(<ItemsCount completedTodosCount={completedTodosCount} todosCount={todosCount}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it ('should render correctly when 2 items left', () => {
    const completedTodosCount = 2;
    const todosCount = 4;
    const wrapper = shallow(<ItemsCount completedTodosCount={completedTodosCount} todosCount={todosCount}/>)
    expect(wrapper).toMatchSnapshot()
  })
})