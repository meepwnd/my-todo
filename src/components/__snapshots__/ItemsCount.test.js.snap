// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`itemscout component with no items should render correctly 1`] = `
<p
  className="item-count"
>
  NaN
   
  items
   left
</p>
`;

exports[`itemscout component with no items should render correctly when 1 item left 1`] = `
<p
  className="item-count"
>
  1
   
  item
   left
</p>
`;

exports[`itemscout component with no items should render correctly when 2 items left 1`] = `
<p
  className="item-count"
>
  2
   
  items
   left
</p>
`;
